patch-package
--- a/node_modules/foundationdb/dist/lib/index.js
+++ b/node_modules/foundationdb/dist/lib/index.js
@@ -26,7 +26,13 @@ function setAPIVersion(version) {
         // We can't support newer versions of fdb without regenerating options & atomics.
         if (version > 510)
             throw Error('The installed version of the FDB Node bindings only support FDB API versions <= 510. Check for updates or file a ticket');
-        native_1.default.setAPIVersion(version);
+        try {
+            native_1.default.setAPIVersion(version);
+        } catch (e) {
+            if (!e || e.message !== 'API version may be set only once') {
+                throw e;
+            }
+        }
         apiVersion = version;
     }
 }
