diff --git a/node_modules/graphql/execution/execute.js b/node_modules/graphql/execution/execute.js
index cf4e628..8dec292 100644
--- a/node_modules/graphql/execution/execute.js
+++ b/node_modules/graphql/execution/execute.js
@@ -725,8 +725,18 @@ function completeObjectValue(exeContext, returnType, fieldNodes, info, path, res
         if (!resolvedIsTypeOf) {
           throw invalidReturnTypeError(returnType, result, fieldNodes);
         }
-
-        return collectAndExecuteSubfields(exeContext, returnType, fieldNodes, path, result);
+        if (returnType.resolveObject) {
+          var resolveObject = returnType.resolveObject(result, exeContext.contextValue, info);
+          if ((0, _isPromise.default)(resolveObject)) {
+            return resolveObject.then(function (resolvedObjectValue) {
+              return collectAndExecuteSubfields(exeContext, returnType, fieldNodes, path, resolvedObjectValue);
+            });
+          } else {
+            return collectAndExecuteSubfields(exeContext, returnType, fieldNodes, path, resolveObject);
+          }
+        } else {
+          return collectAndExecuteSubfields(exeContext, returnType, fieldNodes, path, result);
+        }
       });
     }
 
@@ -735,7 +745,18 @@ function completeObjectValue(exeContext, returnType, fieldNodes, info, path, res
     }
   }
 
-  return collectAndExecuteSubfields(exeContext, returnType, fieldNodes, path, result);
+  if (returnType.resolveObject) {
+    var resolveObject = returnType.resolveObject(result, exeContext.contextValue, info);
+    if ((0, _isPromise.default)(resolveObject)) {
+      return resolveObject.then(function (resolvedObjectValue) {
+        return collectAndExecuteSubfields(exeContext, returnType, fieldNodes, path, resolvedObjectValue);
+      });
+    } else {
+      return collectAndExecuteSubfields(exeContext, returnType, fieldNodes, path, resolveObject);
+    }
+  } else {
+    return collectAndExecuteSubfields(exeContext, returnType, fieldNodes, path, result);
+  }
 }
 
 function invalidReturnTypeError(returnType, result, fieldNodes) {
